partner_relations/model/res_partner_relation_all.py

```diff
@@ -209,4 +210,6 @@ def create(self, vals):
     @api.one
     def unlink(self):
         """divert non-problematic creates to underlying table"""
-        return self.get_underlying_object().unlink()
+        underlying_objs = self.get_underlying_object()
+        underlying_objs = underlying_objs and underlying_objs[0]
+        return underlying_objs.unlink()
```

Comment:
After I've got a shower and re-thought what you wrote, you,re rigth. Here we need a loop or make self.get_underlying_object() returns a recordset not a list.